CDTDatastore/HTTP/CDTIAMSessionCookieInterceptor.m

```diff
@@ -0,0 +1,145 @@
+//
+//  CDTIAMSessionCookieInterceptor.m
+//  CDTDatastore
+//
+//  Created by tomblench on 05/07/2017.
+//  Copyright © 2017 IBM Corporation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "CDTIAMSessionCookieInterceptor.h"
+#import "CDTLogging.h"
+
+static const NSInteger CDTIAMSessionCookieRequestTimeout = 600;
+
+
+@interface CDTIAMSessionCookieInterceptor ()
+
+
+@property NSData *iamSessionRequestBody;
+
+- (NSData *)getBearerToken;
+
+/** NSURLSession to make calls to get IAM bearer token (shouldn't be same one we're intercepting). */
+@property (nonnull, nonatomic, strong) NSURLSession *iamUrlSession;
+
+@property NSURL *iamTokenUrl;
+
+@end
+
+@implementation CDTIAMSessionCookieInterceptor
+
+- (instancetype)initWithAPIKey:(NSString *)apiKey
+{
+    self = [super init];
+    if (self) {
+        // build the request to get the IAM bearer token
+        _iamSessionRequestBody = [[NSString stringWithFormat:@"grant_type=urn:ibm:params:oauth:grant-type:apikey&response_type=cloud_iam&apikey=%@", apiKey] dataUsingEncoding:NSUTF8StringEncoding];
```

Comment:
Given that this worked I guess there aren't any specials here (although I'm surprised `:` isn't), but we should form encode anyway, since we're claiming it is form encoded. I can't remember what the minimum supported level is so not sure whether we can use `stringByAddingPercentEncodingWithAllowedCharacters: [NSCharacterSet URLQueryAllowedCharacterSet]`

It appears this may have been overlooked for the normal `_session` `POST` as well.